src/components/SamlConfiguration/index.test.jsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import Index from './index';
+import LmsApiService from './../../data/services/LmsApiService';
+import { REQUIRED_DATA_FIELDS } from './SamlProviderDataForm';
+
+const formData = new FormData();
+REQUIRED_DATA_FIELDS.forEach(field => formData.append(field, 'testdata'));
+
+describe('<SamlConfiguration /> ', () => {
+  it('creating provider config states', () => {
+    const wrapper = shallow(<Index enterpriseId="testEnterpriseId" />);
+    const spyPostNewProviderConfig = jest.spyOn(LmsApiService, 'postNewProviderConfig');
+
+    const mockResponseData = { data: {} };
+    formData.forEach((value, key) => {
+      mockResponseData.data[key] = value;
+    });
+
+    spyPostNewProviderConfig.mockImplementation(() => mockResponseData);
+    wrapper.instance().createProviderConfig(formData, () => {
+      expect(wrapper.state().providerConfig).toEqual(mockResponseData.data);
+    });
+  });
+
+  it('updating provider config states', () => {
+    const wrapper = shallow(<Index enterpriseId="testEnterpriseId" />);
+    const spyPostUpdatedProviderConfig = jest.spyOn(LmsApiService, 'updateProviderConfig');
+
+    const oldProviderConfigData = { data: {} };
+    const mockResponseData = { data: {} };
+
+    formData.forEach((value, key) => {
+      oldProviderConfigData.data[key] = 'old '.concat(value);
+      mockResponseData.data[key] = value;
+    });
+    wrapper.setState({ providerConfig: oldProviderConfigData });
+
+    spyPostUpdatedProviderConfig.mockImplementation(() => mockResponseData);
+    wrapper.instance().updateProviderConfig(formData, () => {
+      expect(wrapper.state().providerConfig).toEqual(mockResponseData.data);
+    });
+  });
+
+  it('deleting provider config resets states', () => {
+    const wrapper = shallow(<Index enterpriseId="testEnterpriseId" />);
+    const spyPostDeletedProviderConfig = jest.spyOn(LmsApiService, 'deleteProviderConfig');
+
+    const toBeDeletedProviderConfigData = { data: {} };
+    formData.forEach((value, key) => {
+      toBeDeletedProviderConfigData.data[key] = 'old '.concat(value);
+    });
+
+    wrapper.setState({ providerConfig: toBeDeletedProviderConfigData });
+    spyPostDeletedProviderConfig.mockImplementation(() => {});
+    wrapper.instance().deleteProviderConfig(formData, () => {
+      expect(wrapper.state().providerConfig).toEqual(undefined);
+    });
+  });
+
+  it('handling response errors', () => {
```

Comment:
So this particular test isn't super helpful. It is successfully testing that a handler got called, but I think I say it's not helpful because that handler contains logic I'd definitely want to test here if we have a test at all.

There are a couple things you'd need to do to have that functioning though:
1. Mock New Relic calls (there are some examples of that in this repo actually)
2. Use `toHaveReturned()`
3. Possibly change "errorMessage" - I don't know what properties or formatting this JS Error object has.

Reply:
By `change "errorMessage"` do you mean change the hard coded string I have in the test or go about creating 
my own exception differently.?

Reply:
I'm also slightly confused as to what you're saying by "use tohavereturned"

Reply:
So like I said, ideally this test would actually test the logic of the handleErrors which centers entirely around what error status code and message format is brought back.

So changing errorMessage would be to test an API error format, which I think is Json-ish. It might work as is with throwing an Error, but unclear how a JS error here differs from a response JSON blob originating from Python.

And all of the steps below that are what would be needed to test the actual logic. We return a value from handleErrors rather than set state, which is why toHaveReturned() would be used in the assertion.

I don't think fleshing out these tests is blocking or anything and I think it's light enough weight as is to be left. I just worry about either A. someone thinking we have total coverage (when we don't) or B. someone using this as a template of a comprehensive test. I think adding all these tests was probably out of scope for this task to begin with, so I can probably just create a task to flesh out these tests.

Reply:
To my knowledge, `toHaveReturned()` is used to test that a mock function successfully returned, and `toHaveReturnedWith()` is used to to ensure that a mock function returned a specific value. In this test, handleErrors ins't mocked or spied on, so I'm still not sure how to use `toHaveReturned`. If we want to confirm the value returned by `handleErrors` we could possibly do something like 
```
expect(
wrapper.instance().updateProviderConfig(formData)
).resolves.toEqual(errorMessage);
```

Reply:
Ah! My bad. I was thinking we could use it against real functions.

Your suggestion looks apt here for sure.